appcelerator/titanium_mobile
# fix(android): changing picker minDate/maxDate after open

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27625

**Summary:**
Changing picker's `minDate` or `maxDate` after window open would not let user choose value between that range. Would still use the old min/max range. Would also crash if the new range is completely outside of the old range.

**Test:**
1. Build and run the below code on Android.
2. Verify that you can only select a date between May 1st and July 31st.
3. Tap the "Change Min/Max" button.
4. Verify that you can only select a date between January 1st and March 31st.
_(This used to crash before this PR.)_

```javascript
var window = Ti.UI.createWindow({ layout: "vertical" });
var picker = Ti.UI.createPicker({
	type: Ti.UI.PICKER_TYPE_DATE,
	top: "50dp",
	minDate: new Date(2020, 4, 1),
	maxDate: new Date(2020, 6, 31),
});
window.add(picker);
var button = Ti.UI.createButton({ title: "Change Min/Max Date" });
button.addEventListener("click", function() {
	picker.minDate = new Date(2018, 0, 1);
	picker.maxDate = new Date(2018, 2, 31);
});
window.add(button);
window.open();
```
